code/modules/projectiles/gun.dm

```diff
@@ -466,3 +467,12 @@
 
 /obj/item/weapon/gun/proc/bullet_hitting(var/obj/item/projectile/P,var/atom/atarget)
 	return
+
+/obj/item/weapon/gun/kick_act(mob/living/carbon/human/H)
+	. = ..()
+	if(prob(kick_fire_chance))
+		var/list/targets = list()
+		for(var/turf/t in view(6))
```

Comment:
oview is probably better for this so that it doesn't (very confusingly) shoot at its own tile